axelor-business-project/src/main/java/com/axelor/apps/businessproject/service/projectgenerator/factory/ProjectGeneratorFactoryTaskTemplate.java

```diff
@@ -91,20 +93,30 @@ public ActionViewBuilder fill(Project project, SaleOrder saleOrder, LocalDateTim
             .fetchOne();
 
     for (SaleOrderLine orderLine : saleOrder.getSaleOrderLineList()) {
+
+      if (orderLine.getProduct() != null
+          && !(((orderLine.getProduct().getProcurementMethodSelect() != null
```

Comment:
No needs to check "orderLine.getProduct().getProcurementMethodSelect() != null"
Also better create separate variable for product and use it here.